lte/cloud/go/tools/migrations/m013_policy_ipv6/main.go

```diff
@@ -0,0 +1,233 @@
+	if err != nil {
+		return nil, err
+	}
+
+	err = migratePolicyRules(tx, builder)
+	if err != nil {
+		return nil, err
+	}
+
+	if dryRun {
+		return nil, errors.New("throwing error instead of committing because -dry was specified")
+	}
+
+	return nil, nil
+}
+
+// migratePolicyRules derives and creates a policy_qos_profile entity for each
+// existing policy entity.
+// Also removes the Qos field from existing policy entities.
+func migratePolicyRules(tx *sql.Tx, builder squirrel.StatementBuilderType) error {
+	// Get all existing policy ents
+	rows, err := builder.
+		Select(entPkCol, entKeyCol, entNidCol, entGidCol, entConfCol).
+		From(entityTable).
+		Where(squirrel.Eq{entTypeCol: policyEntType}).
+		RunWith(tx).Query()
+	if err != nil {
+		return errors.Wrap(err, "get policy ents")
+	}
+	oldByKey := map[string]policyEnt{}
+	for rows.Next() {
+		var key string
+		old := policyEnt{}
+		var confBytes []byte
+		err = rows.Scan(&old.pk, &key, &old.network, &old.gid, &confBytes)
+		if err != nil {
+			return errors.Wrap(err, "scan policy")
+		}
+		err = json.Unmarshal(confBytes, &old.config)
+		if err != nil {
+			return errors.Wrap(err, "unmarshal existing policy rule config")
+		}
+		oldByKey[key] = old
+	}
+	err = rows.Err()
+	if err != nil {
+		return errors.Wrap(err, "get existing assocs: SQL rows error")
+	}
+
+	// Convert policy configs to {new config}
+	// Note we are explicitly updating the old config to drop the Qos field,
+	// as we use that as the marker of idempotence.
+	updateConfByKey := map[string][]byte{}
+
+	for key, old := range oldByKey {
+		c := old.config
+		newConf := types.PolicyRuleConfig{
+			AppName:        c.AppName,
+			AppServiceType: c.AppServiceType,
+			MonitoringKey:  c.MonitoringKey,
+			Priority:       c.Priority,
+			RatingGroup:    c.RatingGroup,
+			Redirect:       c.Redirect,
+			TrackingType:   c.TrackingType,
+		}
+
+		if c.FlowList != nil {
+			var newFlowList []*types.FlowDescription
+			for _, flow_desc := range c.FlowList {
+				fd := &types.FlowDescription{
+					Action: flow_desc.Action,
+					Match: &types.FlowMatch{
+						Direction: flow_desc.Match.Direction,
+						IPProto:   flow_desc.Match.IPProto,
+						TCPDst: flow_desc.Match.TCPDst,
+						TCPSrc: flow_desc.Match.TCPSrc,
+						UDPDst: flow_desc.Match.UDPDst,
+						UDPSrc: flow_desc.Match.UDPSrc,
+					},
+				}
+				if flow_desc.Match.IPV4Src != "" {
+                    fd.Match.IPSrc = &types.IPAddress{
+                        Version: types.IPAddressVersionIPV4,
+                        Address: flow_desc.Match.IPV4Src,
+                    }
+                }
+				if flow_desc.Match.IPV4Dst != "" {
+                    fd.Match.IPDst = &types.IPAddress{
+                        Version: types.IPAddressVersionIPV4,
+                        Address: flow_desc.Match.IPV4Dst,
+                    }
+                }
+				newFlowList = append(newFlowList, fd)
+			}
+			newConf.FlowList = newFlowList
```

Comment:
your indents are out of control here. also encap this logic in a helper function

you can just mutate `newConf.FlowList` directly as well using `append`